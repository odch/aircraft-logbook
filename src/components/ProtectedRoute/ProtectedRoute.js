import React from 'react'
import PropTypes from 'prop-types'
import { Route, Redirect } from 'react-router-dom'
import { connect } from 'react-redux'

const ProtectedRoute = ({ render, protect, authed, ...rest }) => (
  <Route
    {...rest}
    render={props => {
      if (!protect || authed === true) {
        return render(props)
      }
      return (
        <Redirect
          to={{ pathname: '/login', state: { from: props.location } }}
        />
      )
    }}
  />
)

ProtectedRoute.propTypes = {
  render: PropTypes.func.isRequired,
  protect: PropTypes.bool,
  authed: PropTypes.bool.isRequired,
  location: PropTypes.string
}

const mapStateToProps = state => ({
  authed: !state.firebase.auth.isEmpty
})

const mapActionCreators = {}

export default connect(
  mapStateToProps,
  mapActionCreators
)(ProtectedRoute)
