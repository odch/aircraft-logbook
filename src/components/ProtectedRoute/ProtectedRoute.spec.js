import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import ProtectedRoute from './ProtectedRoute'

const WrappedComponent = () => <div>wrapped</div>

const LoginPage = () => <div>Login page</div>

describe('components', () => {
  describe('ProtectedRoute', () => {
    it('renders component if authenticated', () => {
      const store = configureStore()({
        firebase: {
          auth: {
            isEmpty: false
          }
        }
      })

      const tree = renderer
        .create(
          <Provider store={store}>
            <Router>
              <ProtectedRoute protect render={() => <WrappedComponent />} />
            </Router>
          </Provider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('redirects to login page if not authenticated', () => {
      const store = configureStore()({
        firebase: {
          auth: {
            isEmpty: true
          }
        }
      })

      const tree = renderer
        .create(
          <Provider store={store}>
            <Router>
              <Switch>
                <Route exact path="/login" component={LoginPage} />
                <ProtectedRoute protect render={() => <WrappedComponent />} />
              </Switch>
            </Router>
          </Provider>
        )
        .toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
