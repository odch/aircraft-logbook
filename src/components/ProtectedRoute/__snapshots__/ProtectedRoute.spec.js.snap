// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ProtectedRoute redirects to login page if not authenticated 1`] = `
<div>
  Login page
</div>
`;

exports[`components ProtectedRoute renders component if authenticated 1`] = `
<div>
  wrapped
</div>
`;
